pkg/controllers/cluster/resource/resource.go

```diff
@@ -60,7 +61,17 @@ func MemberServiceForPod(pod *corev1.Pod, cluster *scyllav1.ScyllaCluster) *core
 	if replaceAddr, ok := cluster.Status.Racks[rackName].ReplaceAddressFirstBoot[pod.Name]; ok && replaceAddr != "" {
 		labels[naming.ReplaceLabel] = replaceAddr
 	}
-	return &corev1.Service{
+
+	// No IP can be  found if in pending state and pod should not be replaced
+	if (svc.ObjectMeta.Labels[naming.IpLabel] == "") && pod.Status.Phase == corev1.PodPending {
+		return nil, fmt.Errorf("pod in pending state and service not initialized")
```

Comment:
OPT: I find it a bit detrimental to generate errors for a condition that can happen by design (false alerts). Ideally we should simply log info, and skip create/update of the resource but I understand it would involve a bit of refactoring with caller and may make upstream contribution more difficult.

Reply:
The usage of the error is currently not optional on that design.
In fact we rely on the fact that the reconcile request failed and is put back to the event reconcile queue to finally create with appropriate IP.
If the event doesn't failed it will be deleted which means that we will never go back to this service creation and the associated pod will never be able to get appropriate IP to startup from the service.
I will see if I can change the design/behavior and send a WIP review to dicuss

Reply:
Here is the WIP commit that permitts to avoid relying on that "false" error: https://github.com/criteo-forks/scylla-operator/pull/10/commits/decdd2c5fcc44136fa8599dd426fb9cd562b9d8b